# Cleaning Process
This notebook contain the necessary steps like getting the data info such as data size, number of feature, number of records, mean value, max value, etc. Clean the data which includes dropping some columns and rows, adding more information, joining the dataframes and more

In [204]:
import pandas as pd
import numpy as np
import googlemaps


## Crime Dataset

In [206]:
crime_data = pd.read_csv('csv_files/Crimes_2001_to_Present.csv')

In [207]:
crime_data.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8000075 entries, 0 to 8000074
Data columns (total 22 columns):
 #   Column                Dtype  
---  ------                -----  
 0   ID                    int64  
 1   Case Number           object 
 2   Date                  object 
 3   Block                 object 
 4   IUCR                  object 
 5   Primary Type          object 
 6   Description           object 
 7   Location Description  object 
 8   Arrest                bool   
 9   Domestic              bool   
 10  Beat                  int64  
 11  District              float64
 12  Ward                  float64
 13  Community Area        float64
 14  FBI Code              object 
 15  X Coordinate          float64
 16  Y Coordinate          float64
 17  Year                  int64  
 18  Updated On            object 
 19  Latitude              float64
 20  Longitude             float64
 21  Location              object 
dtypes: bool(2), float64(7), int64(3), object(1

## Converting the Date
#### step 1) changing the data type to a much suitable format

In [208]:
crime_data['Arrest'] = crime_data['Arrest'].replace({
    True:1,
    False:0
})

crime_data['Arrest'].unique()

array([0, 1])

In [209]:

crime_data['New_Date'] = pd.to_datetime(crime_data['Date'])
# crime_data['New_Date'] = crime_data['New_Date'].dt.strftime('%m-%d-%Y %H:%M:%S')

In [210]:
# comparing the old date value format to the datetime value format
crime_data[['Date', 'New_Date']]

Date            New_Date
0        03/18/2015 12:00:00 PM 2015-03-18 12:00:00
1        12/20/2018 03:00:00 PM 2018-12-20 15:00:00
2        05/01/2016 12:25:00 AM 2016-05-01 00:25:00
3        12/20/2018 04:00:00 PM 2018-12-20 16:00:00
4        06/01/2014 12:01:00 AM 2014-06-01 00:01:00
...                         ...                 ...
8000070  08/10/2020 08:30:00 PM 2020-08-10 20:30:00
8000071  06/18/2020 09:00:00 PM 2020-06-18 21:00:00
8000072  07/27/2020 03:02:00 PM 2020-07-27 15:02:00
8000073  08/14/2020 03:00:00 PM 2020-08-14 15:00:00
8000074  03/06/2020 02:00:00 PM 2020-03-06 14:00:00

[8000075 rows x 2 columns]

In [211]:
print("Max new_date value: ", crime_data['New_Date'].max()) # Latest record of 2024
print("Min new_date value: ", crime_data['New_Date'].min()) # Earliest record of 2021

Max new_date value:  2024-02-10 00:00:00
Min new_date value:  2001-01-01 00:00:00


#### Step 2) Dropping the unecessary columns such as X & Y Coordinate, Date, Block, IUCR, Description, Domestic, Beat, District, FBI code, Ward, Updated on, Latitude, Longitude 

In [212]:
col = ['ID', 'New_Date', 'Primary Type', 'Location Description', 'Arrest', 'Community Area', 'Location']
crime_data = crime_data[col]
crime_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8000075 entries, 0 to 8000074
Data columns (total 7 columns):
 #   Column                Dtype         
---  ------                -----         
 0   ID                    int64         
 1   New_Date              datetime64[ns]
 2   Primary Type          object        
 3   Location Description  object        
 4   Arrest                int64         
 5   Community Area        float64       
 6   Location              object        
dtypes: datetime64[ns](1), float64(1), int64(2), object(3)
memory usage: 427.3+ MB


## Step 3) Filtering

### Filter (Pre Covid)
Extracting the data for crime report from 2017-2019

In [213]:
crime_data_2017_2019 = crime_data[(crime_data['New_Date'].dt.year >= 2017) & (crime_data['New_Date'].dt.year <= 2019)]

In [214]:
print("Min new_date value: ", crime_data_2017_2019['New_Date'].min()) # Earliest record of 2017
print("Min new_date value: ", crime_data_2017_2019['New_Date'].max()) # Earliest record of 2019
print(crime_data_2017_2019['New_Date'].dt.year.unique()) # Making sure that the range (2017-2019)

Min new_date value:  2017-01-01 00:00:00
Min new_date value:  2019-12-31 23:55:00
[2018 2019 2017]


In [215]:
crime_data_2017_2019 = crime_data_2017_2019[crime_data_2017_2019['Location'].notna()]
crime_data_2017_2019.reset_index()

index        ID            New_Date                Primary Type  \
0            96  12098557 2019-02-01 00:01:00                     BATTERY   
1           283  12082526 2019-09-24 12:00:00          DECEPTIVE PRACTICE   
2           527  11859264 2019-10-13 06:40:00             CRIMINAL DAMAGE   
3           641  11662417 2019-04-21 12:30:00                     ROBBERY   
4           663  12990873 2019-08-17 13:14:00  OFFENSE INVOLVING CHILDREN   
...         ...       ...                 ...                         ...   
787676  7999292  12022897 2019-12-05 00:00:00          DECEPTIVE PRACTICE   
787677  7999338  12056523 2019-12-20 09:00:00          DECEPTIVE PRACTICE   
787678  7999342  12056116 2019-04-09 00:00:00  OFFENSE INVOLVING CHILDREN   
787679  7999778  12166069 2019-02-01 09:00:00          DECEPTIVE PRACTICE   
787680  7999779  12002967 2018-05-06 16:58:00  OFFENSE INVOLVING CHILDREN   

       Location Description  Arrest  Community Area  \
0                 RESIDENCE       0            63.0   
1                 RESIDENCE       0             3.0   
2                 APARTMENT       0            29.0   
3                 RESIDENCE       0            44.0   
4                 RESIDENCE       1            23.0   
...                     ...     ...             ...   
787676            RESIDENCE       0            34.0   
787677            RESIDENCE       1             8.0   
787678            RESIDENCE       0            48.0   
787679            RESIDENCE       0            16.0   
787680            RESIDENCE       0            30.0   

                             Location  
0       (41.802924631, -87.687367104)  
1       (41.974346203, -87.656361214)  
2       (41.865213688, -87.727590376)  
3         (41.749500329, -87.6011574)  
4        (41.89621515, -87.728572048)  
...                               ...  
787676  (41.837529939, -87.635302767)  
787677  (41.894104719, -87.626221071)  
787678   (41.73099572, -87.563409251)  
787679  (41.951016616, -87.707938347)  
787680  (41.851826431, -87.700973609)  

[787681 rows x 8 columns]

### Filter (Post Covid)
Extracting the data for crime report from 2021 - present

In [216]:
crime_data_2021_present = crime_data[crime_data['New_Date'].dt.year >= 2021]

In [217]:
print("Min new_date value: ", crime_data_2021_present['New_Date'].min()) # Earliest record of 2021
print(crime_data_2021_present['New_Date'].dt.year.unique()) # Making sure that the range (2021-2024)

Min new_date value:  2021-01-01 00:00:00
[2023 2021 2022 2024]


In [218]:

len(crime_data_2021_present['Community Area'].unique())

78

### Step 4) Saving the Dataframe to a CSV file

In [219]:
crime_data_2021_present.to_csv('csv_files/Crimes_2021_to_Present.csv')
crime_data_2017_2019.to_csv('csv_files/Crimes_2017_to_2019.csv')


## Neighborhood Dataset
https://www.zillow.com/research/data/

Contains information like Region Name, State, City, County Name, Average price of a house for every month since 2000

In [220]:
neighborhood_data = pd.read_csv('csv_files/Neighborhood_House_Price.csv')

In [221]:
neighborhood_data.head()

RegionID  SizeRank         RegionName    RegionType StateName State  \
0    112345         0           Maryvale  neighborhood        AZ    AZ   
1    192689         1           Paradise  neighborhood        NV    NV   
2    270958         2    Upper West Side  neighborhood        NY    NY   
3    270957         3    Upper East Side  neighborhood        NY    NY   
4    118208         4  South Los Angeles  neighborhood        CA    CA   

          City                                  Metro          CountyName  \
0      Phoenix              Phoenix-Mesa-Chandler, AZ     Maricopa County   
1    Las Vegas       Las Vegas-Henderson-Paradise, NV        Clark County   
2     New York  New York-Newark-Jersey City, NY-NJ-PA     New York County   
3     New York  New York-Newark-Jersey City, NY-NJ-PA     New York County   
4  Los Angeles     Los Angeles-Long Beach-Anaheim, CA  Los Angeles County   

      2000-01-31  ...    2023-04-30    2023-05-31    2023-06-30    2023-07-31  \
0   66775.313666  ...  3.134925e+05  3.147765e+05  3.166145e+05  3.190725e+05   
1  132638.938818  ...  3.585637e+05  3.580372e+05  3.587546e+05  3.605508e+05   
2  387530.423074  ...  1.276836e+06  1.270266e+06  1.264532e+06  1.258336e+06   
3  634533.128812  ...  1.259968e+06  1.250928e+06  1.245395e+06  1.241081e+06   
4  127876.428774  ...  6.198684e+05  6.208305e+05  6.245314e+05  6.317380e+05   

     2023-08-31    2023-09-30    2023-10-31    2023-11-30    2023-12-31  \
0  3.220546e+05  3.246938e+05  3.271008e+05  3.291411e+05  3.307035e+05   
1  3.634265e+05  3.662741e+05  3.687446e+05  3.708867e+05  3.729634e+05   
2  1.248721e+06  1.238858e+06  1.227969e+06  1.216308e+06  1.208912e+06   
3  1.236655e+06  1.232169e+06  1.224024e+06  1.212976e+06  1.202819e+06   
4  6.413973e+05  6.511754e+05  6.594772e+05  6.659235e+05  6.701266e+05   

     2024-01-31  
0  3.317141e+05  
1  3.748541e+05  
2  1.203406e+06  
3  1.196051e+06  
4  6.678988e+05  

[5 rows x 298 columns]

In [222]:
neighborhood_data = neighborhood_data[(neighborhood_data['State'] == 'IL') & (neighborhood_data['City'] == 'Chicago')]

In [223]:
# removing all the duplicates
neighborhoods_set = set()

for x in neighborhood_data_2021_present['RegionName'].to_list():
    neighborhoods_set.add(x)

len(neighborhoods_set)

181

In [224]:
first_half_column = neighborhood_data.loc[0:, ['RegionID', 'SizeRank', 'RegionName']]
second_half_column =  neighborhood_data.loc[0:, '2017-01-31':'2019-12-31']

first_half_column_2 = neighborhood_data.loc[0:, ['RegionID', 'SizeRank', 'RegionName']]
second_half_column_2 =  neighborhood_data.loc[0:, '2021-01-31':]


In [225]:
neighborhood_data_2017_2019 = pd.concat([first_half_column, second_half_column], axis=1)
neighborhood_data_2021_present = pd.concat([first_half_column_2, second_half_column_2], axis=1)

In [226]:
display(neighborhood_data_2017_2019)
display(neighborhood_data_2021_present)

RegionID  SizeRank               RegionName     2017-01-31  \
42       269589        42                Lake View  320800.529961   
88       403169        88               West Ridge  187637.444136   
97       403117        97           Little Village   90925.075575   
99       269592        99             Logan Square  385189.738477   
123      269590       124             Lincoln Park  527941.394692   
...         ...       ...                      ...            ...   
15991    403329     18535           Heart of Italy  174695.616386   
16031    403380     18592              Golden Gate   25263.130346   
16216    403366     18814                Marycrest  222620.465114   
18723    403369     22019  Mount Greenwood Heights  247078.011675   
20343    403280     24311     Schorsch Forest View  267835.822774   

          2017-02-28     2017-03-31     2017-04-30     2017-05-31  \
42     323403.875393  325554.648715  326994.066372  327962.336934   
88     189553.396344  191027.034117  191889.952161  192260.367968   
97      92159.981171   93255.225637   93859.979012   94812.109531   
99     387715.936437  388969.125310  389250.314752  388709.642645   
123    530161.421716  532967.822467  535106.275721  538274.753360   
...              ...            ...            ...            ...   
15991  177001.337230  178717.892446  180467.268885  181256.703189   
16031   25686.900261   26756.021861   27796.858540   28691.811622   
16216  224828.560250  227139.685233  228359.512801  228371.013902   
18723  248058.672791  248698.293271  248938.921598  249231.377345   
20343  269378.309108  270910.811867  272070.802637  272600.604751   

          2017-06-30     2017-07-31  ...     2019-03-31     2019-04-30  \
42     327811.478035  327962.790422  ...  327436.112620  327638.943873   
88     191657.654310  191412.149087  ...  204783.405502  204814.591194   
97      95124.032869   95401.547996  ...  124782.842092  126055.982656   
99     387398.503138  386883.966756  ...  414528.089528  416111.230615   
123    540584.689695  544013.398935  ...  533113.718875  531744.025668   
...              ...            ...  ...            ...            ...   
15991  181725.536626  182351.215235  ...  211387.283107  211614.709182   
16031   29155.295553   29502.305780  ...   41621.830434   41514.056903   
16216  226888.913337  225638.915889  ...  239572.563303  239978.803273   
18723  249169.393812  249098.501465  ...  258493.304020  258668.042545   
20343  271903.447634  271082.505246  ...  282809.435451  283935.866960   

          2019-05-31     2019-06-30     2019-07-31     2019-08-31  \
42     327428.128541  326810.851395  325654.374039  324470.264123   
88     204405.009083  204249.895972  203823.411224  203506.384146   
97     126805.514295  127412.094949  126981.804065  127537.226453   
99     415797.190256  414317.801661  412092.450971  410604.326047   
123    528767.619898  527003.180168  525522.950173  524547.076555   
...              ...            ...            ...            ...   
15991  210971.351293  210925.979425  210781.085712  210695.122221   
16031   42116.313584   42738.846464   43075.210579   43220.250096   
16216  240549.150432  240891.286584  240668.961675  240444.077082   
18723  258522.936241  259107.675903  259501.419125  260679.817344   
20343  283764.886591  283086.990139  281545.591961  280313.131720   

          2019-09-30     2019-10-31     2019-11-30     2019-12-31  
42     323266.430150  322409.458344  321944.598794  322010.510163  
88     202946.678331  203030.557504  203433.095567  204670.639264  
97     127830.794507  128503.409806  129227.471761  130157.108534  
99     409346.866443  408934.379163  409406.784465  411298.166343  
123    522445.253179  521778.078281  520769.321925  520116.842387  
...              ...            ...            ...            ...  
15991  209817.312211  209887.855266  210484.309940  212598.767758  
16031   43592.471256   44341.406856   44976.630856   45283.533368  
16216  240122.75090

RegionID  SizeRank               RegionName     2021-01-31  \
42       269589        42                Lake View  335152.964276   
88       403169        88               West Ridge  213600.229920   
97       403117        97           Little Village  150609.690221   
99       269592        99             Logan Square  438745.604328   
123      269590       124             Lincoln Park  521486.137219   
...         ...       ...                      ...            ...   
15991    403329     18535           Heart of Italy  230912.584221   
16031    403380     18592              Golden Gate   55617.851577   
16216    403366     18814                Marycrest  275760.178768   
18723    403369     22019  Mount Greenwood Heights  280946.442648   
20343    403280     24311     Schorsch Forest View  307632.094045   

          2021-02-28     2021-03-31     2021-04-30     2021-05-31  \
42     337100.070376  338529.196964  339798.571621  340754.044648   
88     215838.540172  217654.905330  219105.803791  219738.822606   
97     154491.459007  159032.561218  163509.685324  166556.653383   
99     442793.484892  446040.066569  448541.706261  449749.184842   
123    522856.818945  523647.182240  523534.460888  522886.530447   
...              ...            ...            ...            ...   
15991  233706.648789  235914.918863  236975.399000  237436.862080   
16031   57571.195221   60538.071205   63795.056935   66521.929432   
16216  280995.201316  286359.511962  290788.403519  294977.065756   
18723  285493.773982  289930.026637  293810.501845  297387.517453   
20343  311662.776952  316208.501575  320674.893231  323826.163279   

          2021-06-30     2021-07-31  ...     2023-04-30     2023-05-31  \
42     341788.369111  343067.924135  ...  345791.989131  348243.716515   
88     220295.245610  221026.998730  ...  219505.468935  220873.395671   
97     168400.346553  169456.453588  ...  168529.731016  170215.552104   
99     449763.281273  450720.802504  ...  438170.581717  441159.953902   
123    522981.141822  524273.725310  ...  537512.792193  539139.274169   
...              ...            ...  ...            ...            ...   
15991  238114.885444  241895.703793  ...  272724.187575  277481.545154   
16031   68435.029445   69220.692911  ...   70320.634522   69738.047004   
16216  299432.029132  303617.607781  ...  317947.364806  318242.748000   
18723  301473.582012  304741.517336  ...  310257.672150  311483.749658   
20343  325993.122187  326625.966628  ...  332068.430625  333953.617594   

          2023-06-30     2023-07-31     2023-08-31     2023-09-30  \
42     351331.955928  354372.109448  358274.505322  361767.239038   
88     222693.533150  224815.395489  227233.660863  228789.686683   
97     171405.516615  172511.518632  174512.511548  176942.109290   
99     444235.237026  446830.281993  449198.964839  449859.667570   
123    541588.440708  543950.923591  546390.818324  548270.857647   
...              ...            ...            ...            ...   
15991  281176.120792  284109.937646  287336.590620  289204.659588   
16031   68916.978047   67902.767071   66675.576774   65910.205072   
16216  319146.595262  320886.278715  322325.703420  322872.760639   
18723  313023.103028  314335.600736  315228.320421  315700.383051   
20343  337033.145114  341258.354083  345718.141282  349568.777665   

          2023-10-31     2023-11-30     2023-12-31     2024-01-31  
42     364245.140803  365404.235809  365273.873903  364719.918837  
88     229263.487182  229190.543498  229300.289538  229288.170544  
97     178664.607033  179329.071751  179274.978734  179381.863942  
99     449269.546622  448226.586436  447136.438126  446721.925355  
123    549318.705650  549895.692694  549323.122838  548638.640501  
...              ...            ...            ...            ...  
15991  289954.458620  289743.066667  290485.432481  291655.160235  
16031   65409.901479   65276.616920   64771.155875   64550.671104  
16216  322881.47220

## Google Cloud Provider
using GCP to get all 181 neighborhood top left and bottom right coordinates
This will be useful when we will compare house prices with crime rate based on neighborhoods, and the most safest neighborhood in chicago

In [ ]:
pip install googlemaps

In [227]:
with open('api_key.txt', 'r') as f:
    api_key = f.read().replace('\n','')
api_key

'AIzaSyDneplBDeuLGH84jqZZYz0o6APJBFcL1ek'

### Step 1) Getting the coordinates from Google and saving it to a variable

In [229]:

gmaps = googlemaps.Client(key=api_key)

def geocode_location(location):
    # Geocode the location
    try:
        geocode_result = gmaps.geocode(location)
        print(geocode_result)

        lat = geocode_result[0]['geometry']['location']['lat']
        lng = geocode_result[0]['geometry']['location']['lng']
        return lat, lng
    except Exception as e:
        print("An error occured while getting the API")
    
def get_bounding_box(center, distance):

    distance_in_degrees = distance / 111111  # 1 degree = 111,111 km, so 
    
    # Calculate top left and bottom right coordinates
    top_left = (center[0] + distance_in_degrees, center[1] - distance_in_degrees)
    bottom_right = (center[0] - distance_in_degrees, center[1] + distance_in_degrees)
    
    return top_left, bottom_right


# location = 'Little Village, Chicago'

loc_list = neighborhoods_set
coordinates_map_1 = {}
coordinates_map_2 = {}
for loc in loc_list:
    full_loc = loc + ', Chicago'
    center = geocode_location(full_loc) # get the api 
    # print("original", center[0], center[1]) # checking purposes

    if center is not None:
        distance = 1000 # boundary distance
        top_left, bottom_right = get_bounding_box(center, distance)
        # print(f"Top Left: {top_left}")
        # print(f"Bottom Right: {bottom_right}")
        coordinates_map_1[loc] = top_left
        coordinates_map_2[loc] = bottom_right
    else:
        print("Geocoding failed.")


[{'address_components': [{'long_name': 'West Beverly', 'short_name': 'West Beverly', 'types': ['neighborhood', 'political']}, {'long_name': 'Chicago', 'short_name': 'Chicago', 'types': ['locality', 'political']}, {'long_name': 'Worth Township', 'short_name': 'Worth Township', 'types': ['administrative_area_level_3', 'political']}, {'long_name': 'Cook County', 'short_name': 'Cook County', 'types': ['administrative_area_level_2', 'political']}, {'long_name': 'Illinois', 'short_name': 'IL', 'types': ['administrative_area_level_1', 'political']}, {'long_name': 'United States', 'short_name': 'US', 'types': ['country', 'political']}], 'formatted_address': 'West Beverly, Chicago, IL, USA', 'geometry': {'bounds': {'northeast': {'lat': 41.7136897, 'lng': -87.6813805}, 'southwest': {'lat': 41.699048, 'lng': -87.69647599999999}}, 'location': {'lat': 41.7056889, 'lng': -87.6866751}, 'location_type': 'APPROXIMATE', 'viewport': {'northeast': {'lat': 41.7136897, 'lng': -87.6813805}, 'southwest': {'la

### Step 2) Turning the variable which contain the coordinates into a dataframe. Had to create 2 dataframe (one for top left coordinate, and another one for bottom right coordinate

In [230]:
temp_df_1 = pd.DataFrame(coordinates_map_1)
temp_df_1 = temp_df_1.rename(index= {0:'TL_Lat', 1:'TL_Long'})

temp_df_2 = pd.DataFrame(coordinates_map_2)
temp_df_2 = temp_df_2.rename(index= {0:'BR_Lat', 1:'BR_Long'})


In [231]:
display(temp_df_1)
display(temp_df_2)

West Beverly  Ravenswood Manor  Dearborn Park  South Commons  \
TL_Lat      41.714689         41.973889      41.875429      41.851183   
TL_Long    -87.695675        -87.710250     -87.638000     -87.629337   

         Belmont Central  Ford City  Marycrest  New Eastside  Kelvin Park  \
TL_Lat         41.936463  41.765747  41.746604     41.895240    41.941349   
TL_Long       -87.787303 -87.741495 -87.714872    -87.626421   -87.753337   

          Roseland  ...  Hanson Park  West Elsdon  Ravenswood Gardens  \
TL_Lat   41.719834  ...    41.930280    41.802219           41.973741   
TL_Long -87.632583  ...   -87.769964   -87.732146          -87.702858   

         Noble Square  Bronzeville     Cragin  \
TL_Lat      41.908331    41.825644  41.940294   
TL_Long    -87.672851   -87.625774 -87.775255   

         University Village - Little Italy  LeClaire Courts    Ashburn  \
TL_Lat                           41.878540        41.819345  41.758690   
TL_Long                         -87.660133       -87.758952 -87.721007   

         Wicker Park  
TL_Lat     41.919489  
TL_Long   -87.686567  

[2 rows x 181 columns]

West Beverly  Ravenswood Manor  Dearborn Park  South Commons  \
BR_Lat      41.696689         41.955889      41.857429      41.833182   
BR_Long    -87.677675        -87.692250     -87.619999     -87.611337   

         Belmont Central  Ford City  Marycrest  New Eastside  Kelvin Park  \
BR_Lat         41.918463  41.747747  41.728604     41.877240    41.923349   
BR_Long       -87.769302 -87.723495 -87.696872    -87.608421   -87.735337   

          Roseland  ...  Hanson Park  West Elsdon  Ravenswood Gardens  \
BR_Lat   41.701834  ...    41.912280    41.784219           41.955741   
BR_Long -87.614583  ...   -87.751964   -87.714146          -87.684858   

         Noble Square  Bronzeville     Cragin  \
BR_Lat      41.890331    41.807644  41.922293   
BR_Long    -87.654851   -87.607774 -87.757255   

         University Village - Little Italy  LeClaire Courts    Ashburn  \
BR_Lat                           41.860540        41.801344  41.740690   
BR_Long                         -87.642133       -87.740952 -87.703007   

         Wicker Park  
BR_Lat     41.901489  
BR_Long   -87.668567  

[2 rows x 181 columns]

In [232]:
temp_df_1 = temp_df_1.transpose() # transpose
temp_df_2 = temp_df_2.transpose()

In [233]:
display(temp_df_1)
display(temp_df_2)

TL_Lat    TL_Long
West Beverly                       41.714689 -87.695675
Ravenswood Manor                   41.973889 -87.710250
Dearborn Park                      41.875429 -87.638000
South Commons                      41.851183 -87.629337
Belmont Central                    41.936463 -87.787303
...                                      ...        ...
Cragin                             41.940294 -87.775255
University Village - Little Italy  41.878540 -87.660133
LeClaire Courts                    41.819345 -87.758952
Ashburn                            41.758690 -87.721007
Wicker Park                        41.919489 -87.686567

[181 rows x 2 columns]

BR_Lat    BR_Long
West Beverly                       41.696689 -87.677675
Ravenswood Manor                   41.955889 -87.692250
Dearborn Park                      41.857429 -87.619999
South Commons                      41.833182 -87.611337
Belmont Central                    41.918463 -87.769302
...                                      ...        ...
Cragin                             41.922293 -87.757255
University Village - Little Italy  41.860540 -87.642133
LeClaire Courts                    41.801344 -87.740952
Ashburn                            41.740690 -87.703007
Wicker Park                        41.901489 -87.668567

[181 rows x 2 columns]

In [234]:
temp_df_1 = temp_df_1.reset_index()
temp_df_2 = temp_df_2.reset_index()


In [235]:
display(temp_df_1)
display(temp_df_2)

index     TL_Lat    TL_Long
0                         West Beverly  41.714689 -87.695675
1                     Ravenswood Manor  41.973889 -87.710250
2                        Dearborn Park  41.875429 -87.638000
3                        South Commons  41.851183 -87.629337
4                      Belmont Central  41.936463 -87.787303
..                                 ...        ...        ...
176                             Cragin  41.940294 -87.775255
177  University Village - Little Italy  41.878540 -87.660133
178                    LeClaire Courts  41.819345 -87.758952
179                            Ashburn  41.758690 -87.721007
180                        Wicker Park  41.919489 -87.686567

[181 rows x 3 columns]

index     BR_Lat    BR_Long
0                         West Beverly  41.696689 -87.677675
1                     Ravenswood Manor  41.955889 -87.692250
2                        Dearborn Park  41.857429 -87.619999
3                        South Commons  41.833182 -87.611337
4                      Belmont Central  41.918463 -87.769302
..                                 ...        ...        ...
176                             Cragin  41.922293 -87.757255
177  University Village - Little Italy  41.860540 -87.642133
178                    LeClaire Courts  41.801344 -87.740952
179                            Ashburn  41.740690 -87.703007
180                        Wicker Park  41.901489 -87.668567

[181 rows x 3 columns]

### Step 3 Merge the two dataframe


In [236]:
neighborhood_coordinates = temp_df_1.merge(temp_df_2, left_on='index', right_on='index')
neighborhood_coordinates.rename(columns={'index': 'RegionName'}, inplace=True)


In [237]:
neighborhood_coordinates

RegionName     TL_Lat    TL_Long     BR_Lat  \
0                         West Beverly  41.714689 -87.695675  41.696689   
1                     Ravenswood Manor  41.973889 -87.710250  41.955889   
2                        Dearborn Park  41.875429 -87.638000  41.857429   
3                        South Commons  41.851183 -87.629337  41.833182   
4                      Belmont Central  41.936463 -87.787303  41.918463   
..                                 ...        ...        ...        ...   
176                             Cragin  41.940294 -87.775255  41.922293   
177  University Village - Little Italy  41.878540 -87.660133  41.860540   
178                    LeClaire Courts  41.819345 -87.758952  41.801344   
179                            Ashburn  41.758690 -87.721007  41.740690   
180                        Wicker Park  41.919489 -87.686567  41.901489   

       BR_Long  
0   -87.677675  
1   -87.692250  
2   -87.619999  
3   -87.611337  
4   -87.769302  
..         ...  
176 -87.757255  
177 -87.642133  
178 -87.740952  
179 -87.703007  
180 -87.668567  

[181 rows x 5 columns]

### Step 5) Adding neighborhood_coordinates to the neighborhood house data

In [238]:
display(neighborhood_data_2017_2019)
display(neighborhood_data_2021_present)

RegionID  SizeRank               RegionName     2017-01-31  \
42       269589        42                Lake View  320800.529961   
88       403169        88               West Ridge  187637.444136   
97       403117        97           Little Village   90925.075575   
99       269592        99             Logan Square  385189.738477   
123      269590       124             Lincoln Park  527941.394692   
...         ...       ...                      ...            ...   
15991    403329     18535           Heart of Italy  174695.616386   
16031    403380     18592              Golden Gate   25263.130346   
16216    403366     18814                Marycrest  222620.465114   
18723    403369     22019  Mount Greenwood Heights  247078.011675   
20343    403280     24311     Schorsch Forest View  267835.822774   

          2017-02-28     2017-03-31     2017-04-30     2017-05-31  \
42     323403.875393  325554.648715  326994.066372  327962.336934   
88     189553.396344  191027.034117  191889.952161  192260.367968   
97      92159.981171   93255.225637   93859.979012   94812.109531   
99     387715.936437  388969.125310  389250.314752  388709.642645   
123    530161.421716  532967.822467  535106.275721  538274.753360   
...              ...            ...            ...            ...   
15991  177001.337230  178717.892446  180467.268885  181256.703189   
16031   25686.900261   26756.021861   27796.858540   28691.811622   
16216  224828.560250  227139.685233  228359.512801  228371.013902   
18723  248058.672791  248698.293271  248938.921598  249231.377345   
20343  269378.309108  270910.811867  272070.802637  272600.604751   

          2017-06-30     2017-07-31  ...     2019-03-31     2019-04-30  \
42     327811.478035  327962.790422  ...  327436.112620  327638.943873   
88     191657.654310  191412.149087  ...  204783.405502  204814.591194   
97      95124.032869   95401.547996  ...  124782.842092  126055.982656   
99     387398.503138  386883.966756  ...  414528.089528  416111.230615   
123    540584.689695  544013.398935  ...  533113.718875  531744.025668   
...              ...            ...  ...            ...            ...   
15991  181725.536626  182351.215235  ...  211387.283107  211614.709182   
16031   29155.295553   29502.305780  ...   41621.830434   41514.056903   
16216  226888.913337  225638.915889  ...  239572.563303  239978.803273   
18723  249169.393812  249098.501465  ...  258493.304020  258668.042545   
20343  271903.447634  271082.505246  ...  282809.435451  283935.866960   

          2019-05-31     2019-06-30     2019-07-31     2019-08-31  \
42     327428.128541  326810.851395  325654.374039  324470.264123   
88     204405.009083  204249.895972  203823.411224  203506.384146   
97     126805.514295  127412.094949  126981.804065  127537.226453   
99     415797.190256  414317.801661  412092.450971  410604.326047   
123    528767.619898  527003.180168  525522.950173  524547.076555   
...              ...            ...            ...            ...   
15991  210971.351293  210925.979425  210781.085712  210695.122221   
16031   42116.313584   42738.846464   43075.210579   43220.250096   
16216  240549.150432  240891.286584  240668.961675  240444.077082   
18723  258522.936241  259107.675903  259501.419125  260679.817344   
20343  283764.886591  283086.990139  281545.591961  280313.131720   

          2019-09-30     2019-10-31     2019-11-30     2019-12-31  
42     323266.430150  322409.458344  321944.598794  322010.510163  
88     202946.678331  203030.557504  203433.095567  204670.639264  
97     127830.794507  128503.409806  129227.471761  130157.108534  
99     409346.866443  408934.379163  409406.784465  411298.166343  
123    522445.253179  521778.078281  520769.321925  520116.842387  
...              ...            ...            ...            ...  
15991  209817.312211  209887.855266  210484.309940  212598.767758  
16031   43592.471256   44341.406856   44976.630856   45283.533368  
16216  240122.75090

RegionID  SizeRank               RegionName     2021-01-31  \
42       269589        42                Lake View  335152.964276   
88       403169        88               West Ridge  213600.229920   
97       403117        97           Little Village  150609.690221   
99       269592        99             Logan Square  438745.604328   
123      269590       124             Lincoln Park  521486.137219   
...         ...       ...                      ...            ...   
15991    403329     18535           Heart of Italy  230912.584221   
16031    403380     18592              Golden Gate   55617.851577   
16216    403366     18814                Marycrest  275760.178768   
18723    403369     22019  Mount Greenwood Heights  280946.442648   
20343    403280     24311     Schorsch Forest View  307632.094045   

          2021-02-28     2021-03-31     2021-04-30     2021-05-31  \
42     337100.070376  338529.196964  339798.571621  340754.044648   
88     215838.540172  217654.905330  219105.803791  219738.822606   
97     154491.459007  159032.561218  163509.685324  166556.653383   
99     442793.484892  446040.066569  448541.706261  449749.184842   
123    522856.818945  523647.182240  523534.460888  522886.530447   
...              ...            ...            ...            ...   
15991  233706.648789  235914.918863  236975.399000  237436.862080   
16031   57571.195221   60538.071205   63795.056935   66521.929432   
16216  280995.201316  286359.511962  290788.403519  294977.065756   
18723  285493.773982  289930.026637  293810.501845  297387.517453   
20343  311662.776952  316208.501575  320674.893231  323826.163279   

          2021-06-30     2021-07-31  ...     2023-04-30     2023-05-31  \
42     341788.369111  343067.924135  ...  345791.989131  348243.716515   
88     220295.245610  221026.998730  ...  219505.468935  220873.395671   
97     168400.346553  169456.453588  ...  168529.731016  170215.552104   
99     449763.281273  450720.802504  ...  438170.581717  441159.953902   
123    522981.141822  524273.725310  ...  537512.792193  539139.274169   
...              ...            ...  ...            ...            ...   
15991  238114.885444  241895.703793  ...  272724.187575  277481.545154   
16031   68435.029445   69220.692911  ...   70320.634522   69738.047004   
16216  299432.029132  303617.607781  ...  317947.364806  318242.748000   
18723  301473.582012  304741.517336  ...  310257.672150  311483.749658   
20343  325993.122187  326625.966628  ...  332068.430625  333953.617594   

          2023-06-30     2023-07-31     2023-08-31     2023-09-30  \
42     351331.955928  354372.109448  358274.505322  361767.239038   
88     222693.533150  224815.395489  227233.660863  228789.686683   
97     171405.516615  172511.518632  174512.511548  176942.109290   
99     444235.237026  446830.281993  449198.964839  449859.667570   
123    541588.440708  543950.923591  546390.818324  548270.857647   
...              ...            ...            ...            ...   
15991  281176.120792  284109.937646  287336.590620  289204.659588   
16031   68916.978047   67902.767071   66675.576774   65910.205072   
16216  319146.595262  320886.278715  322325.703420  322872.760639   
18723  313023.103028  314335.600736  315228.320421  315700.383051   
20343  337033.145114  341258.354083  345718.141282  349568.777665   

          2023-10-31     2023-11-30     2023-12-31     2024-01-31  
42     364245.140803  365404.235809  365273.873903  364719.918837  
88     229263.487182  229190.543498  229300.289538  229288.170544  
97     178664.607033  179329.071751  179274.978734  179381.863942  
99     449269.546622  448226.586436  447136.438126  446721.925355  
123    549318.705650  549895.692694  549323.122838  548638.640501  
...              ...            ...            ...            ...  
15991  289954.458620  289743.066667  290485.432481  291655.160235  
16031   65409.901479   65276.616920   64771.155875   64550.671104  
16216  322881.47220

In [239]:
neighborhood_data_2017_2019 = neighborhood_data_2017_2019.merge(neighborhood_coordinates, left_on='RegionName', right_on='RegionName')
neighborhood_data_2021_present = neighborhood_data_2021_present.merge(neighborhood_coordinates, left_on='RegionName', right_on='RegionName')


In [240]:
display(neighborhood_data_2017_2019)
display(neighborhood_data_2021_present)

RegionID  SizeRank               RegionName     2017-01-31  \
0      269589        42                Lake View  320800.529961   
1      403169        88               West Ridge  187637.444136   
2      403117        97           Little Village   90925.075575   
3      269592        99             Logan Square  385189.738477   
4      269590       124             Lincoln Park  527941.394692   
..        ...       ...                      ...            ...   
176    403329     18535           Heart of Italy  174695.616386   
177    403380     18592              Golden Gate   25263.130346   
178    403366     18814                Marycrest  222620.465114   
179    403369     22019  Mount Greenwood Heights  247078.011675   
180    403280     24311     Schorsch Forest View  267835.822774   

        2017-02-28     2017-03-31     2017-04-30     2017-05-31  \
0    323403.875393  325554.648715  326994.066372  327962.336934   
1    189553.396344  191027.034117  191889.952161  192260.367968   
2     92159.981171   93255.225637   93859.979012   94812.109531   
3    387715.936437  388969.125310  389250.314752  388709.642645   
4    530161.421716  532967.822467  535106.275721  538274.753360   
..             ...            ...            ...            ...   
176  177001.337230  178717.892446  180467.268885  181256.703189   
177   25686.900261   26756.021861   27796.858540   28691.811622   
178  224828.560250  227139.685233  228359.512801  228371.013902   
179  248058.672791  248698.293271  248938.921598  249231.377345   
180  269378.309108  270910.811867  272070.802637  272600.604751   

        2017-06-30     2017-07-31  ...     2019-07-31     2019-08-31  \
0    327811.478035  327962.790422  ...  325654.374039  324470.264123   
1    191657.654310  191412.149087  ...  203823.411224  203506.384146   
2     95124.032869   95401.547996  ...  126981.804065  127537.226453   
3    387398.503138  386883.966756  ...  412092.450971  410604.326047   
4    540584.689695  544013.398935  ...  525522.950173  524547.076555   
..             ...            ...  ...            ...            ...   
176  181725.536626  182351.215235  ...  210781.085712  210695.122221   
177   29155.295553   29502.305780  ...   43075.210579   43220.250096   
178  226888.913337  225638.915889  ...  240668.961675  240444.077082   
179  249169.393812  249098.501465  ...  259501.419125  260679.817344   
180  271903.447634  271082.505246  ...  281545.591961  280313.131720   

        2019-09-30     2019-10-31     2019-11-30     2019-12-31     TL_Lat  \
0    323266.430150  322409.458344  321944.598794  322010.510163  41.948781   
1    202946.678331  203030.557504  203433.095567  204670.639264  42.009580   
2    127830.794507  128503.409806  129227.471761  130157.108534  41.853527   
3    409346.866443  408934.379163  409406.784465  411298.166343  41.932060   
4    522445.253179  521778.078281  520769.321925  520116.842387  41.934470   
..             ...            ...            ...            ...        ...   
176  209817.312211  209887.855266  210484.309940  212598.767758  41.857614   
177   43592.471256   44341.406856   44976.630856   45283.533368  41.663165   
178  240122.750905  240710.728818  241326.823178  242843.243896  41.746604   
179  260902.811661  261775.682500  262210.501428  262427.216995  41.700590   
180  278990.521479  278549.676198  278265.908508  279195.960665  41.971647   

       TL_Long     BR_Lat    BR_Long  
0   -87.667927  41.930781 -87.649927  
1   -87.701577  41.991580 -87.683577  
2   -87.714045  41.835527 -87.696045  
3   -87.718291  41.914060 -87.700291  
4   -87.657778  41.916470 -87.639778  
..         ...        ...        ...  
176 -87.693616  41.839614 -87.675616  
177 -87.617195  41.645165 -87.599195  
178 -87.714872  41.728604 -87.696872  
179 -87.709805  41.682590 -87.691805  
180 -87.850469  41.953647 -87.832469  

[181 rows x 43 columns]

RegionID  SizeRank               RegionName     2021-01-31  \
0      269589        42                Lake View  335152.964276   
1      403169        88               West Ridge  213600.229920   
2      403117        97           Little Village  150609.690221   
3      269592        99             Logan Square  438745.604328   
4      269590       124             Lincoln Park  521486.137219   
..        ...       ...                      ...            ...   
176    403329     18535           Heart of Italy  230912.584221   
177    403380     18592              Golden Gate   55617.851577   
178    403366     18814                Marycrest  275760.178768   
179    403369     22019  Mount Greenwood Heights  280946.442648   
180    403280     24311     Schorsch Forest View  307632.094045   

        2021-02-28     2021-03-31     2021-04-30     2021-05-31  \
0    337100.070376  338529.196964  339798.571621  340754.044648   
1    215838.540172  217654.905330  219105.803791  219738.822606   
2    154491.459007  159032.561218  163509.685324  166556.653383   
3    442793.484892  446040.066569  448541.706261  449749.184842   
4    522856.818945  523647.182240  523534.460888  522886.530447   
..             ...            ...            ...            ...   
176  233706.648789  235914.918863  236975.399000  237436.862080   
177   57571.195221   60538.071205   63795.056935   66521.929432   
178  280995.201316  286359.511962  290788.403519  294977.065756   
179  285493.773982  289930.026637  293810.501845  297387.517453   
180  311662.776952  316208.501575  320674.893231  323826.163279   

        2021-06-30     2021-07-31  ...     2023-08-31     2023-09-30  \
0    341788.369111  343067.924135  ...  358274.505322  361767.239038   
1    220295.245610  221026.998730  ...  227233.660863  228789.686683   
2    168400.346553  169456.453588  ...  174512.511548  176942.109290   
3    449763.281273  450720.802504  ...  449198.964839  449859.667570   
4    522981.141822  524273.725310  ...  546390.818324  548270.857647   
..             ...            ...  ...            ...            ...   
176  238114.885444  241895.703793  ...  287336.590620  289204.659588   
177   68435.029445   69220.692911  ...   66675.576774   65910.205072   
178  299432.029132  303617.607781  ...  322325.703420  322872.760639   
179  301473.582012  304741.517336  ...  315228.320421  315700.383051   
180  325993.122187  326625.966628  ...  345718.141282  349568.777665   

        2023-10-31     2023-11-30     2023-12-31     2024-01-31     TL_Lat  \
0    364245.140803  365404.235809  365273.873903  364719.918837  41.948781   
1    229263.487182  229190.543498  229300.289538  229288.170544  42.009580   
2    178664.607033  179329.071751  179274.978734  179381.863942  41.853527   
3    449269.546622  448226.586436  447136.438126  446721.925355  41.932060   
4    549318.705650  549895.692694  549323.122838  548638.640501  41.934470   
..             ...            ...            ...            ...        ...   
176  289954.458620  289743.066667  290485.432481  291655.160235  41.857614   
177   65409.901479   65276.616920   64771.155875   64550.671104  41.663165   
178  322881.472206  322911.150672  323036.432989  323600.645730  41.746604   
179  316257.417595  316753.787873  317284.417069  317851.664055  41.700590   
180  352424.332892  354896.000629  356164.834436  356733.602104  41.971647   

       TL_Long     BR_Lat    BR_Long  
0   -87.667927  41.930781 -87.649927  
1   -87.701577  41.991580 -87.683577  
2   -87.714045  41.835527 -87.696045  
3   -87.718291  41.914060 -87.700291  
4   -87.657778  41.916470 -87.639778  
..         ...        ...        ...  
176 -87.693616  41.839614 -87.675616  
177 -87.617195  41.645165 -87.599195  
178 -87.714872  41.728604 -87.696872  
179 -87.709805  41.682590 -87.691805  
180 -87.850469  41.953647 -87.832469  

[181 rows x 44 columns]

### Step 6) Saving all 3 dataframe to cvs


In [241]:
neighborhood_coordinates.to_csv('csv_files/neighborhood_coordinates.csv')
neighborhood_data_2017_2019.to_csv('csv_files/neighborhood_data_2017_2019.csv')
neighborhood_data_2021_present.to_csv('csv_files/neighborhood_data_2021_present.csv')


<H1><I>END OF CLEANING PROCESS<I><H1>